openshift/openshift-ansible-contrib
# Fix node label customisation

#### What does this PR do?
This PR fixes problems with label customisation.

- In order for the heat templates to work, it adds an empty dictionary to both infra and app groups if there isn't any label defined for them.
- The `openshift_cluster_node_labels` definition is moved to `all.yml` due to its usage in the `openstack-stack` role during provisioning.
- Documentation for customising node labels is expanded.

#### How should this be manually tested?
1) In `all.yml`, add custom definition to `openshift_cluster_node_labels` which contains only one of the available groups (app or infra).
2) Run provisioning with `-vvv` option and take a look at the output of the tasks right before the `openstack-stack` role (in particular, `Set openshift_cluster_node_labels for the app group if not defined`). It should show the content of `openshift_cluster_node_labels` variable that has both `app` and `infra` keys in it (along with your custom label).
3) Check that the end to end deployment is successful.
4) Connect to the OpenShift master node and display all nodes, including their labels with `oc get nodes --show-labels`. Your custom labels should be printed out along with the predefined labels.

#### Is there a relevant Issue open for this?
Closes #674 

#### Who would you like to review this?
cc: @tomassedovic @bogdando PTAL
